[docs] EAS Build - clarify npm/yarn configuration (#12775)
expo/expo: docs/pages/build-reference/how-tos.md

## How to use private package repositories

> - Configure your project in a way that relies on the `NPM_TOKEN` env variable to authenticate with private repositories

Reviewer: ```suggestion
- Configure your project in a way that relies on the `NPM_TOKEN` env variable to authenticate with private repositories.
```

<ImageSpotlight alt="Secret creation UI filled" src="/static/images/eas-build/environment-secrets/secrets-create-filled.png" />

> Before setting up private packages, check exisitng configuration (`.npmrc` and `yarnrc.yml`) described on [Server infrastructure page](/build-reference/infrastructure), to verify if it's not affecting your setup.

Reviewer: ```suggestion
Before setting up private packages, check the existing configuration (`.npmrc` and `yarnrc.yml`) described on the [Build server infrastructure](/build-reference/infrastructure) page, to verify if it's not affecting your setup.
```

> Before setting up private packages, check the existing configuration (`.npmrc` and `yarnrc.yml`) described on the [Build server infrastructure](/build-reference/infrastructure) page, to verify if it's not affecting your setup.

Reviewer 2: ```suggestion
Before setting up private packages, check the existing configuration described in the [build server infrastructure](/build-reference/infrastructure.md) page under `.npmrc` and `yarnrc.yml` to verify that it won't affect your setup.
```

I'm guessing that anyone setting this up would understand what will/won't affect their setup? I haven't used private packages so that was my only question when reading this

Author: yes, I also updated the first line of the instruction to make it clearer that user needs to know how to configure their project

btw `.md` extensions in link in markdown are not necessary, I'm not sure if they work)

Reviewer 2: they should work (just tested), both live on docs.expo.io as well as clicking to navigate in VSCode
